shared/testutil/block.go

```diff
@@ -155,52 +155,149 @@ func GenerateFullBlock(
 	return &ethpb.SignedBeaconBlock{Block: block, Signature: signature.Marshal()}, nil
 }
 
+// GenerateProposerSlashingForValidator for a specific validator index.
+func GenerateProposerSlashingForValidator(
+	bState *stateTrie.BeaconState,
+	priv *bls.SecretKey,
+	idx uint64,
+) (*ethpb.ProposerSlashing, error) {
+	header1 := &ethpb.SignedBeaconBlockHeader{
+		Header: &ethpb.BeaconBlockHeader{
+			Slot:     bState.Slot(),
+			BodyRoot: []byte{0, 1, 0},
+		},
+	}
+	root, err := ssz.HashTreeRoot(header1.Header)
+	if err != nil {
+		return nil, err
+	}
+	currentEpoch := helpers.CurrentEpoch(bState)
+	domain := helpers.Domain(bState.Fork(), currentEpoch, params.BeaconConfig().DomainBeaconProposer)
+	header1.Signature = priv.Sign(root[:], domain).Marshal()
+
+	header2 := &ethpb.SignedBeaconBlockHeader{
+		Header: &ethpb.BeaconBlockHeader{
+			Slot:     bState.Slot(),
+			BodyRoot: []byte{0, 2, 0},
+		},
+	}
+	root, err = ssz.HashTreeRoot(header2.Header)
+	if err != nil {
+		return nil, err
+	}
+	header2.Signature = priv.Sign(root[:], domain).Marshal()
+
+	return &ethpb.ProposerSlashing{
+		ProposerIndex: idx,
+		Header_1:      header1,
+		Header_2:      header2,
+	}, nil
+}
+
 func generateProposerSlashings(
 	bState *stateTrie.BeaconState,
 	privs []*bls.SecretKey,
 	numSlashings uint64,
 ) ([]*ethpb.ProposerSlashing, error) {
-	currentEpoch := helpers.CurrentEpoch(bState)
-
 	proposerSlashings := make([]*ethpb.ProposerSlashing, numSlashings)
 	for i := uint64(0); i < numSlashings; i++ {
 		proposerIndex, err := randValIndex(bState)
 		if err != nil {
 			return nil, err
 		}
-		header1 := &ethpb.SignedBeaconBlockHeader{
-			Header: &ethpb.BeaconBlockHeader{
-				Slot:     bState.Slot(),
-				BodyRoot: []byte{0, 1, 0},
-			},
-		}
-		root, err := ssz.HashTreeRoot(header1.Header)
+		slashing, err := GenerateProposerSlashingForValidator(bState, privs[proposerIndex], proposerIndex)
 		if err != nil {
 			return nil, err
 		}
-		domain := helpers.Domain(bState.Fork(), currentEpoch, params.BeaconConfig().DomainBeaconProposer)
-		header1.Signature = privs[proposerIndex].Sign(root[:], domain).Marshal()
+		proposerSlashings[i] = slashing
+	}
+	return proposerSlashings, nil
+}
 
-		header2 := &ethpb.SignedBeaconBlockHeader{
-			Header: &ethpb.BeaconBlockHeader{
-				Slot:     bState.Slot(),
-				BodyRoot: []byte{0, 2, 0},
-			},
-		}
-		root, err = ssz.HashTreeRoot(header2.Header)
-		if err != nil {
-			return nil, err
+// GenerateAttesterSlashingForValidator for a specific validator index.
+func GenerateAttesterSlashingForValidator(
+	bState *stateTrie.BeaconState,
```

Comment:
i think attester slashing is much less restrictive.
maybe you dont need the  assignments and committees 
if we create indexed attestation instead and sign a slashable data it should be enough to be slashable
https://github.com/prysmaticlabs/prysm/blob/ecfd7bdfa118fac1224e4d9de559fd358c3b0fb6/beacon-chain/core/blocks/block_operations.go#L500
anyway your way also works